### ISSUE TITLE
[Docs]: Role Based Access Control (RBAC) is outdated/wrong

### ISSUE THREAD

**Issue by mk-webfleet** [cloudnative-pg/cloudnative-pg#5377]
### Is there an existing issue already for your request/idea?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new documentation enhancement to be evaluated.

### What problem in the existing documentation this issue aims to solve?

The problem is that the current documentation states something about the cluster role cnpg-manager. (https://cloudnative-pg.io/documentation/1.24/security/#role-based-access-control-rbac). But this role is not deployed with the current helm chart (https://github.com/cloudnative-pg/charts/blob/main/charts/cloudnative-pg/templates/rbac.yaml).

It would be good to update the section of the documentation to give administrators the chance to have correct arguments when talking to their security department to get approval to run the operator.

### Additional context

Our security department reviews all new operators which require additional permissions which are not limited to its own namespace. Giving wrong information to them makes a bad first impression and reduces the chance of getting the operator deployed. Also, reading through the complete source code is not an option.

### Backport?

Yes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by sxd** [comment]
Hello @mk-webfleet 

Can you clarify which role is the one that is not deployed?

Thank you!

**Comment by mk-webfleet** [comment]
Hi @sxd,

currently, the role cnpg-manager is not deployed. But the following roles are deployed:
* cnpg-cloudnative-pg
* cnpg-cloudnative-pg-edit
* cnpg-cloudnative-pg-view

See https://github.com/cloudnative-pg/charts/blob/main/charts/cloudnative-pg/templates/rbac.yaml#L21, https://github.com/cloudnative-pg/charts/blob/main/charts/cloudnative-pg/templates/rbac.yaml#L393 and https://github.com/cloudnative-pg/charts/blob/main/charts/cloudnative-pg/templates/rbac.yaml#L417

Thx for investigating.

**Comment by sxd** [comment]
@mk-webfleet 
The documentation states  "The specific name and type of this role also depend on the deployment method", probably here is missing documentation for the chart but not the operator, probably we can ask the creator of that patch to document the roles in the helm chart?
On the documentation side I don't see a problem, since we cannot document every possible way to name a role or type of it

Regards,

**Comment by mk-webfleet** [comment]
@sxd: I agree with you that the name can be different. But what puzzled me was the fact that instead of one role, now 3 roles are deployed. And since the helm chart is part of this project I thought I mention it. But if you think that this is out of scope, then please forgive me and close the ticket with "wont do". Fine with me. Thx anyway for your time.

**Comment by sxd** [comment]
@mk-webfleet the helm chart it's another project, same organization but different project, so it should have it's own documentation I think, probably we should add this documentation there too with the specific modification for that, what do you think about it? 
What do you think about adding documentation to the helm chart @phisco ? 

Thanks for bringing this @mk-webfleet it's really useful to have the perspective of users !

**Comment by mk-webfleet** [comment]
@sxd: I think my confusion came from the fact that https://cloudnative-pg.io/documentation/1.24/ is the primary site for documentation. Also, in a sub-page for installation (https://cloudnative-pg.io/documentation/1.24/installation_upgrade/#using-the-helm-chart) one method is Helm. Nowhere is stated that depending on the installation method, there are differences for the operator.

From the documentation page of view, the installation method is just a tiny part of the whole page and as a user I would expect that regardless how I get the operator running, everything else in the documentation applies.
